src/inc/CombFilter.h

```diff
@@ -0,0 +1,108 @@
+#pragma once
+
+//IIR class CCombFilterBase;
+#include <math.h>
+#include <algorithm> 
+#include <complex>
+using namespace std;
+#include "CombFilterIf.h"
+
+class CIIR
+{
+private:
+	float m_fMaxDelayLengthInS;
+	float m_fSampleRateInHz;
+	int m_iNumChannels;
+	int m_M;
+	float *m_pDelayline;
+public:
+	CIIR(float fMaxDelayLengthInS, float fSampleRateInHz, int iNumChannels);
+	~CIIR();
+	void reset();
+	void filter(float data_in[], float data_out[], int len);
+	Error_t process(float **ppfInputBuffer, float **ppfOutputBuffer, int iNumberOfFrames, int iNumChannel);
+};
+class CFIR
+{
+private:
+	float m_fMaxDelayLengthInS;
+	float m_fSampleRateInHz;
+	int m_iNumChannels;
+	int m_M;
+	float *m_pDelayline;
+public:
+	CFIR(float fMaxDelayLengthInS, float fSampleRateInHz, int iNumChannels);
```

Comment:
Class CIIR and CFIR can ideally be the child classes to combfilterbase as the only difference is in the filter process, the initializations can be handled by the base class. Using inheritance here will result in cleaner code and reduce redundancy, however this approach works as well!